text/0000-long-error-codes-explanation-normalization.md

```diff
@@ -0,0 +1,116 @@
+
+Start Date: 2016-01-04
+
+RFC PR:
+
+Rust Issue: N/A
+
+# Summary
+
+Long error codes explanations haven't been normalized yet. This RFC intends to do it in order to uniformize them.
+
+# Motivation
+
+Long error codes explanations are a very important part of Rust. Having an explanation of what failed helps to understand the error and is appreciated by Rust developers of all skill levels. Providing an unified template is needed in order to help people who would want to write ones as well as people who read them.
+
+# Detailed design
+
+Here is the template I propose:
+
+## Error description
+
+Giving a little more detailed error message. For example, the `E0109` says "type parameters are not allowed on this type" and the error explanation says: "You tried to give a type parameter to a type which doesn't need it.".
+
```

Comment:
This example could be better since the extended description is not any more descriptive than the actual error message. And since the extended message is the same length as the error message, if it was better we would just make it the error message.